http/oidc/src/main/java/org/wildfly/security/http/oidc/TokenValidator.java

```diff
@@ -83,11 +85,18 @@ private TokenValidator(Builder builder) {
      * @throws OidcException if the ID token is invalid
      */
     public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken) throws OidcException {
+        return parseAndVerifyToken(idToken, accessToken, null);
+    }
+
+    public VerifiedTokens parseAndVerifyToken(final String idToken, final String accessToken, final String cookiesStr) throws OidcException {
         try {
             JwtContext idJwtContext = setVerificationKey(idToken, jwtConsumerBuilder);
             jwtConsumerBuilder.setExpectedAudience(clientConfiguration.getResourceName());
             jwtConsumerBuilder.registerValidator(new AzpValidator(clientConfiguration.getResourceName()));
             jwtConsumerBuilder.registerValidator(new AtHashValidator(accessToken, clientConfiguration.getTokenSignatureAlgorithm()));
+            if (cookiesStr != null) {
+                jwtConsumerBuilder.registerValidator(new NonceValidator(cookiesStr));
```

Comment:
The `if` here shouldn't be needed. Since we are now always sending the nonce in the authentication request, this validator should always be registered.